File: backend/app/schemas/feed.py
import uuid
from datetime import datetime
from pydantic import BaseModel
from app.schemas.movie import MoviePublic
from app.schemas.user import UserPublic


class FeedItem(BaseModel):
    id: uuid.UUID
    user: UserPublic
    movie: MoviePublic
    score: float | None
    review: str | None
    contains_spoiler: bool
    watched_on: str | None
    created_at: datetime
    like_count: int = 0
    liked_by_me: bool = False

    model_config = {"from_attributes": True}


class FeedPage(BaseModel):
    items: list[FeedItem]
    next_cursor: str | None
    has_more: bool

File: frontend/src/types/feed.ts
import type { User } from './user'
import type { Movie } from './movie'

export interface FeedItem {
  id: string
  user: User
  movie: Movie
  score: number | null
  review: string | null
  contains_spoiler: boolean
  watched_on: string | null
  created_at: string
  like_count: number
  liked_by_me: boolean
}

export interface FeedPage {
  items: FeedItem[]
  next_cursor: string | null
  has_more: boolean
}
